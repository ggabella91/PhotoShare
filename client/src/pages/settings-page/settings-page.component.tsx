import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { FormInput } from '../../components/form-input/form-input.component';
import Button from '../../components/button/button.component';
import Alert from 'react-bootstrap/Alert';

import './settings-page.styles.scss';

const SettingsPage: React.FC = () => {
  const [userInfo, setUserInfo] = useState({
    name: '',
    email: '',
  });

  const [userPassword, setUserPassword] = useState({
    passwordCurrent: '',
    password: '',
    passwordConfirm: '',
  });

  const { name, email } = userInfo;
  const { passwordCurrent, password, passwordConfirm } = userPassword;

  const handleInfoChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value, name } = event.target;

    setUserInfo({ ...userInfo, [name]: value });
  };

  const handleSubmitInfo = async (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    // changeInfoStart(name, email);
  };

  const handlePasswordChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value, name } = event.target;

    setUserPassword({ ...userPassword, [name]: value });
  };

  const handleSubmitPassword = async (
    event: React.FormEvent<HTMLFormElement>
  ) => {
    event.preventDefault();

    // changePasswordStart(passwordCurrent, password, passwordConfirm);
  };

  return (
    <div className='settings'>
      <h2>Settings</h2>
      <form className='change-info' onSubmit={handleSubmitInfo}>
        <span>Update your info</span>
        <FormInput
          type='text'
          name='name'
          value={name}
          onChange={handleInfoChange}
          label='name'
        />
        <FormInput
          type='email'
          name='email'
          value={email}
          onChange={handleInfoChange}
          label='email'
        />
        <div className='button'>
          <Button
            className='submit-button settings-button'
            onSubmit={handleSubmitInfo}
          >
            <span className='update-info'>Update Info</span>
          </Button>
        </div>
      </form>
      <form className='change-info' onSubmit={handleSubmitPassword}>
        <span>Change your password</span>
        <FormInput
          type='password'
          name='passwordCurrent'
          value={passwordCurrent}
          onChange={handlePasswordChange}
          label='current password'
        />
        <FormInput
          type='password'
          name='password'
          value={password}
          onChange={handlePasswordChange}
          label='new password'
        />
        <FormInput
          type='password'
          name='passwordConfirm'
          value={passwordConfirm}
          onChange={handlePasswordChange}
          label='confirm new password'
        />
        <div className='button'>
          <Button
            className='submit-button settings-button password-button'
            onSubmit={handleSubmitPassword}
          >
            <span className='update-info password'>Change Password</span>
          </Button>
        </div>
      </form>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({});

const mapDispatchProps = () => ({});

export default connect(mapStateToProps, mapDispatchProps)(SettingsPage);
