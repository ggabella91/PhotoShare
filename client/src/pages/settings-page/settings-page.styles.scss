.settings {
  width: 850px;
  height: 1000px;
  display: flex;
  flex-direction: column;
  justify-items: center;
  align-items: center;
  margin-top: 30px;
  margin-left: 50%;
  transform: translate(-50%);

  .update-profile-photo {
    margin-top: 20px;

    .profile-photo-container {
      margin-top: 15px;
      width: 300px;
      height: 300px;
      border: solid 2px;
      border-radius: 150px;
    }

    span {
      font-weight: 400;
      margin: 0px;
      width: 300px;
      line-height: 1.5;
    }
  }

  .group {
    margin: 10px;
  }

  .change-info {
    min-width: 200px;
    max-width: 250px;
    top: 20px;
    display: flex;
    flex-direction: column;
    justify-items: center;
    align-items: center;
    justify-content: center;
    margin: 40px 0px 40px 0px;

    span {
      font-weight: 400;
      margin: 0px;
      width: 300px;
      line-height: 1.5;
    }

    .update-info {
      margin-bottom: 50%;
      transform: translateY(50%);
    }

    .password {
      transform: translateY(15%);
    }

    .password-button {
      min-height: 80px;
    }
  }

  .settings-button {
    min-width: 180px;
    max-width: 220px;

    span {
      font-weight: 400;
    }
  }
}
