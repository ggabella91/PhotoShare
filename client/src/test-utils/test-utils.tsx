import { FC, ReactElement } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, RenderOptions } from '@testing-library/react';
import { store } from '../redux/store';

interface WrapperProps {
  wrapperProps?: { route: string; location: string };
}

type CustomRenderOptions = RenderOptions & WrapperProps;

const WithReduxProvider: FC<WrapperProps> = ({ children, wrapperProps }) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <Routes>
          <Route path={wrapperProps?.route || '/'} element={children} />
        </Routes>
      </BrowserRouter>
    </Provider>
  );
};

const renderWithReduxAndRouter = (
  ui: ReactElement,
  options?: Omit<CustomRenderOptions, 'wrapper'>
) => {
  window.history.pushState({}, '', options?.wrapperProps?.location || '/');
  const customContainer = document.createElement('div');
  customContainer.id = 'root';

  return render(ui, {
    wrapper: (props) => (
      <WithReduxProvider {...props} wrapperProps={options?.wrapperProps} />
    ),
    container: document.body.appendChild(customContainer),
    ...options,
  });
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { renderWithReduxAndRouter as render };
