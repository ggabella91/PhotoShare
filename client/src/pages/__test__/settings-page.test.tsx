import { render } from '../../test-utils/test-utils';
import { SettingsPage } from '../settings-page/settings-page.component';

it('renders a settings page component', () => {
  const { container: settingsPageWrapper } = render(<SettingsPage />);

  expect(settingsPageWrapper).toBeInTheDocument();
});
