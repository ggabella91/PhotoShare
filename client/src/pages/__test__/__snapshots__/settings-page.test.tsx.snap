// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a settings page component 1`] = `
<Tabs
  id="settings-page"
  mountOnEnter={false}
  transition={false}
  unmountOnExit={false}
  variant="tabs"
>
  <Tab
    eventKey="update-profile-photo"
    title="Update Profile Photo"
  >
    <Connect(UpdateProfilePhoto) />
  </Tab>
  <Tab
    className="settings"
    eventKey="update-info"
    title="Update Info"
  >
    <Connect(UpdateInfo) />
  </Tab>
  <Tab
    className="settings"
    eventKey="update-password"
    title="Update Password"
  >
    <Connect(UpdatePassword) />
  </Tab>
</Tabs>
`;
